splink/blocking_rule_library.py

```diff
@@ -0,0 +1,169 @@
+    ):
+        super().__init__(
+            salting_partitions=salting_partitions, arrays_to_explode=arrays_to_explode
+        )
+        self.sql_condition = sql_condition
+
+        self.base_dialect_str = base_dialect_str
+
+    def create_sql(self, sql_dialect: SplinkDialect) -> str:
+        sql_condition = self.sql_condition
+        if self.base_dialect_str is not None:
+            base_dialect = SplinkDialect.from_string(self.base_dialect_str)
+            # if we are told it is one dialect, but try to create comparison level
+            # of another, try to translate with sqlglot
+            if sql_dialect != base_dialect:
+                base_dialect_sqlglot_name = base_dialect.sqlglot_name
+
+                # as default, translate condition into our dialect
+                try:
+                    sql_condition = _translate_sql_string(
+                        sql_condition,
+                        sql_dialect.sqlglot_name,
+                        base_dialect_sqlglot_name,
+                    )
+                # if we hit a sqlglot error, assume users knows what they are doing,
+                # e.g. it is something custom / unknown to sqlglot
+                # error will just appear when they try to use it
+                except TokenError:
+                    pass
+        return sql_condition
+
+
+class _Merge(BlockingRuleCreator):
+    @final
+    def __init__(
+        self,
+        *blocking_rules: Union[BlockingRuleCreator, dict],
+        salting_partitions=None,
+        arrays_to_explode=None,
+    ):
+        super().__init__(
+            salting_partitions=salting_partitions, arrays_to_explode=arrays_to_explode
+        )
+        num_levels = len(blocking_rules)
+        if num_levels == 0:
+            raise ValueError(
+                f"Must provide at least one blocking rule to {type(self)}()"
+            )
+        self.blocking_rules = blocking_rules
+
+    @property
+    def salting_partitions(self):
+        if (
+            hasattr(self, "_salting_partitions")
+            and self._salting_partitions is not None
+        ):
+            return self._salting_partitions
+
+        return max(
+            [
+                br.salting_partitions
+                for br in self.blocking_rules
+                if br.salting_partitions is not None
+            ],
+            default=None,
+        )
+
+    @property
+    def arrays_to_explode(self):
+        if hasattr(self, "_arrays_to_explode"):
+            return self._arrays_to_explode
+
+        if any([br.arrays_to_explode for br in self.blocking_rules]):
+            raise ValueError("Cannot merge blocking rules with arrays_to_explode")
+        return None
+
+    @final
+    def create_sql(self, sql_dialect: SplinkDialect) -> str:
+        return f" {self._clause} ".join(
+            map(lambda cl: f"({cl.create_sql(sql_dialect)})", self.blocking_rules)
+        )
+
+
+class And(_Merge):
+    _clause = "AND"
+
+
+class Or(_Merge):
```

Comment:
I left `Or` in to get tests passing, but i think there's an argument for dropping support for it:  it's very rare you want to use an OR in a blocking rule because it usually results in a full cartesian product of the input dataframe being created